src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,93 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import duke.commands.*;
+import duke.tasks.*;
+
+/**
+ * The Parser class encapsulates the parsing of user input and file text when loading from the hard disk.
+ */
+public class Parser {
+    /**
+     * Parses the given input to determine the type of command the input contains.
+     * @param input Input to be parsed.
+     * @return Command object
+     * @throws DukeException
+     */
+    public static Command parseInput(String input) throws DukeException{
+        String[] splitInput = input.split(" ", 2);
+        String command = splitInput[0].trim();
+        String detail = "";
+        Boolean description_needed = command.equals("todo") || command.equals("deadline") || command.equals("event") ||
```

Comment:
> Avoid Complicated Expressions

Instead of checking the command, perhaps you check the negation since you already have a default case below. Since you are only checking for list perhaps use `boolean hasDescription = !command.equals("list");`
However to account for the checking of missing description, perhaps you can actually abstract out CommandWithDescription from the various commands and throw the missing description error there as an exception?